controllers/user.js

```diff
@@ -27,24 +28,110 @@ router.get( '/:id', (req, res, next) => {
 
 
 router.put( '/:id', (req, res, next) => {
-  req.user.update( req.body, (error, result) => {
+  // mask data (only pass on qualifying values)
+  const updateData = {};
+  if ( req.body.level ) updateData.level = req.body.level;
+  if ( req.body.newbie === false || req.body.newbie === true ) updateData.newbie = req.body.newbie;
+  if ( req.body.name ) updateData.name = req.body.name;
+
+  req.user.update( updateData, (error, result) => {
     if ( error ) return next(error);
     res.json(result);
   });
 });
 
 
 
-router.get( '/:id/lexemes',
-  (req, res, next) => {
-    // get list of IDs of user's words
-    User.findById( req.user._id )
-      .then( user => {
-        const ids = user.lexemes.map( lexeme => lexeme.lexeme_id )
-        console.log(ids);
-      })
-      .catch( error => next(error) );
+router.get( '/:id/lexemes', (req, res, next) => {
+  User.findById( req.user._id ) // get again b/c in req.user the lexemes are stripped off
+    .populate('lexemes.lexeme', 'lexeme')
+    .then( ({ lexemes }) => res.json(lexemes) )
+    .catch( error => next(error) );
+});
+
+
+router.put( '/:id/lexemes', (req, res, next) => {
+
+  // move to models.js?
+  User.findById( req.user._id )
```

Comment:
Yes you could put each Model into a file and add all the logic related to the db there, your routes will look much cleaner.
Controller should take care of preparing the data to be used on the Repository layer(model)